subsys/bluetooth/audio/pbp.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/types.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/audio/audio.h>
+#include <zephyr/bluetooth/audio/pbp.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/check.h>
+
+LOG_MODULE_REGISTER(bt_pbp, CONFIG_BT_PBP_LOG_LEVEL);
+
+/* PBA Service UUID + Public Broadcast Announcement features + Public_Info metadata of length 1 */
+#define MIN_PBA_SIZE			5
+
+int bt_pbp_get_announcement(const uint8_t meta[], size_t meta_len,
+			    enum bt_pbp_announcement_feature features,
+			    struct net_buf_simple *pba_data_buf)
+{
+	CHECKIF(pba_data_buf == NULL) {
+		LOG_DBG("No buffer provided for advertising data!\n");
+
+		return -EINVAL;
+	}
+
+	CHECKIF((meta == NULL && meta_len != 0) || (meta != NULL && meta_len == 0)) {
+		LOG_DBG("Invalid metadata combination: %p %zu", meta, meta_len);
+
+		return -EINVAL;
+	}
+
+	CHECKIF(pba_data_buf->size < meta_len + 3) {
+		LOG_DBG("Buffer size needs to be at least %d!\n", meta_len + 3);
+
+		return -EINVAL;
```

Comment:
What is the magic number `3` here? 
Can we replace it with a #define or some `sizeof`? 

I assume it's related to the 16-bit UUID and the 8-bit features, but shouldn't it also always have the 8-bit metadata length field as well? I.e. shouldn't the size be minimum `2 + 1 + 1 + meta_len`?